Trimble.Modus.Components/Controls/TMNumberInput/TMNumberInput.xaml.cs

```diff
@@ -0,0 +1,231 @@
+    {
+        TMNumberInput numberInput = bindable as TMNumberInput;
+        double max = (double)newValue;
+        double min = numberInput.MinValue;
+        if (min > max)
+        {
+            throw new ArgumentException("Min value cannot be greater than max value");
+        }
+        else if (numberInput.Value > max)
+        {
+            numberInput.Value = max;
+        }
+        numberInput.TMInputControl.ToggleRightIconState(numberInput.Value < max);
+    }
+
+    /// <summary>
+    /// Called when the <see cref="Value"/> property is changed.
+    /// </summary>
+    private static void OnValueChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable is TMNumberInput numberInput)
+        {
+            numberInput.TMInputControl.Text = newValue.ToString();
+        }
+    }
+
+    /// <summary>
+    /// Method to validate that the input is a number and that it is within the min and max values
+    /// </summary>
+    private async void OnTextChanged(object sender, TextChangedEventArgs e)
+    {
+        string newText = e.NewTextValue;
+        string oldText = e.OldTextValue;
+        TMInput tMInput = sender as TMInput;
+        tMInput.TextChanged -= OnTextChanged;
+        if (string.IsNullOrEmpty(newText) || newText == "-" || newText == "." || newText == "-." || newText == "," || newText == "-,")
+        {
+            tMInput.TextChanged += OnTextChanged;
+            return;
+        }
+
+        // This is a hack since the TextChanged event is fired again when we set the text
+        await Task.Delay(10);
+        if (double.TryParse(newText, out double number))
+        {
+            if (number > MaxValue)
+            {
+                tMInput.Text = MaxValue.ToString();
+                ValueChanged?.Invoke(this, new ValueChangedEventArgs(number, MaxValue));
+            }
+            else if (number < MinValue)
+            {
+                tMInput.Text = MinValue.ToString();
+                ValueChanged?.Invoke(this, new ValueChangedEventArgs(number, MinValue));
+            }
+        }
+        else
+        {
+            tMInput.Text = oldText;
+        }
+        tMInput.ToggleRightIconState(IsEnabled && double.Parse(tMInput.Text) < MaxValue);
+        tMInput.ToggleLeftIconState(IsEnabled && double.Parse(tMInput.Text) > MinValue);
+        tMInput.TextChanged += OnTextChanged;
+    }
+
+    /// <summary>
+    /// Command for left minus icon
+    /// </summary>
+    private ICommand MinusCommand
+    {
+        get
+        {
+            return _minusCommand ?? (_minusCommand = new Command(() =>
+            {
+                if (double.TryParse(TMInputControl.Text, out double number))
+                {
+                    if (number > MinValue)
+                    {
+                        number -= Step;
+                        ValueChanged?.Invoke(this, new ValueChangedEventArgs(number + Step, number));
```

Comment:
`ValueChanged?.Invoke` and `TMInputControl.Text` are meant to be called together and they are repeated 4 times in this class. Can you move them to a method? Like `UpdateValue(oldvalue, newvalue)`?

Reply:
We should also have a 'ValueChangedCommand' exposed, we can have that as well inside this new method.

Reply:
Done